### PR TITLE
operator: Use PodAntiAffinity for all components

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Based on previous discussions on #9503 this PR replaces TopologySpreadConstraints with PodAntiAffinity rules for the `lokistack-gateway`, `distirbutor` and `querier` deployments when trying to spread them across nodes. In addition it adds the missing default PodAntiAffinity for the `index-gateway` component.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: operator/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Main
 
+- [9930](https://github.com/grafana/loki/pull/9930) **periklis**: Use PodAntiAffinity for all components
```
Comment (xperimental) [inline operator/CHANGELOG.md:3]: I forgot to add the changelog line for my PR, can we fix this here? (don't know if this needs a changelog entry though. It fixed an issue that had not appeared yet.

```suggestion
- [9930](https://github.com/grafana/loki/pull/9930) **periklis**: Use PodAntiAffinity for all components
- [9860](https://github.com/grafana/loki/pull/9860) **xperimental**: Fix update of labels and annotations of PodTemplates
```

File: operator/internal/manifests/var.go [hunk 2]
```diff
@@ -567,11 +549,6 @@ func defaultNodeAffinity(enableNodeAffinity bool) *corev1.NodeAffinity {
 // defaultPodAntiAffinity for components in podAntiAffinityComponents will
 // configure pods, of a LokiStack, to preferably not run on the same node
 func defaultPodAntiAffinity(componentLabel, stackName string) *corev1.PodAntiAffinity {
```
Comment (xperimental) [inline operator/internal/manifests/var.go:551]: Sorry, can not comment on the correct lines:

---

(in `configureAffinity`, L514-516):

```go
	if cSpec.PodAntiAffinity != nil {
		affinity.PodAntiAffinity = cSpec.PodAntiAffinity
	}
```

Should we "merge" the default with the custom PodAntiAffinity or is overwriting with the custom one enough?

---

(in `configureAffinity`, L518-520):

```go
	if affinity.NodeAffinity == nil && affinity.PodAntiAffinity == nil {
		return nil
	}
```

If I am not mistaken, then we now _always_ have a PodAntiAffinity, so the function will never return `nil`. I think we do not need the if-block anymore.

Comment (periklis) [reply operator/internal/manifests/var.go:547]: 1. We should overwrite, that's enough IMHO.
2. You are right the if branch is not needed anymore.

**Review by xperimental — COMMENTED** [review]
LGTM.

On a second read-through I noticed that it might be an idea to split the tests for podAntiAffinity into "default" and "custom" test cases to avoid confusion (maybe it's just me). Have a look at [this commit](https://github.com/xperimental/loki/commit/515553c7e2eb2e950e7f774a031822ae115da3bf). It's optional though, consider this an "approve".

**Review by xperimental — APPROVED** [review]

**Pull request merged**